From 5090603d01779bb1717fb0c50953330e8770550f Mon Sep 17 00:00:00 2001
From: Pierre-Jean Texier <pjtexier@koncepto.io>
Date: Sun, 23 Jun 2019 14:11:37 +0200
Subject: [PATCH] fix install of binaries with a static only library

Define CMAKE_INSTALL_RPATH only if ENABLE_SHARED is set otherwise the
following issue occured:

CMake Error at iiod/cmake_install.cmake:45 (file):
  file RPATH_CHANGE could not write new RPATH:

    /usr/lib

  to the file:

    /home/pjtexier/<path>/usr/sbin/iiod

  No valid ELF RPATH or RUNPATH entry exists in the file; File does not have
  a valid ELF identification.
Call Stack (most recent call first):
  cmake_install.cmake:50 (include)

Signed-off-by: Pierre-Jean Texier <pjtexier@koncepto.io>
[Upstream status: https://github.com/analogdevicesinc/libiio/pull/283]
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d9b7ea1..634e553 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,7 @@ endif()
 
 set(CMAKE_INSTALL_DOCDIR "" CACHE PATH "documentation root (DATAROOTDIR/doc/${PROJECT_NAME}${LIBIIO_VERSION_MAJOR}-doc)")
 include(GNUInstallDirs)
-if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+if(ENABLE_SHARED AND ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 	set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_FULL_LIBDIR}")
 	set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 endif()
-- 
2.7.4

